### PR TITLE
Modernize android and support RN 0.59

### PR DESCRIPTION
**Pull request by kdawgwilk**
This modernizes the android side of things and bumps Lottie android to 2.8 in order to support androidX and RN 0.59 that is now using SDK 28+

### CODE REVIEW

**Comment by kdawgwilk** [comment]
CI is being weird and not seeing installed `node_modules`

**Comment by emilioicai** [comment]
Hey @kdawgwilk  , thanks for this PR! I've been trying myself to update lottie-android for quite some time so it's great you managed to do it. I'll take a look at the CI error to see if we can unblock this

**Comment by lorenc-tomasz** [comment]
Why commited files within the  npm-packages-offline-cache directory?

**Comment by lorenc-tomasz** [comment]
Connected with https://github.com/react-native-community/lottie-react-native/issues/453

I have resolve the issue temporary with adding the api("com.android.support:appcompat-v7:28.0.0") to gradle

**Comment by emilioicai** [comment]
@kdawgwilk maybe there's an issue in your yarn.lock file? can you try removing it?

**Comment by yurykorzun** [comment]
Because I constantly have problems with gradle files in my RN dependencies, I override it with a fixed gradle file in settings.gradle

`def gradleLottieTemplateText = file(rootProject.projectDir.getAbsolutePath() + '/lottie.gradle.template').text
project(':lottie-react-native').getBuildFile().write(gradleLottieTemplateText)`

**Comment by kdawgwilk** [comment]
@lorenc-tomasz I also wondered about the committed npm folder but when I added it to the `.gitignore` they marked as deleted so I figured the maintainer wanted them there ¯\_(ツ)_/¯

File: example/android/app/build.gradle [hunk 1]
```diff
@@ -129,7 +126,7 @@ android {
         variant.outputs.each { output ->
             // For each separate APK per architecture, set a unique version code as described here:
             // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
-            def versionCodes = ["armeabi-v7a":1, "x86":2]
+            def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86-64": 4]
```
Comment (bondparkerbond) [inline example/android/app/build.gradle]: @kdawgwilk I think x86-64 was a typo in the React Native repo, as they changed it in todays .59.1 bugfix version to x86_64. Can we change the two instances of this or is there a reason we wouldn't want to change it?
See https://github.com/facebook/react-native/commit/4b996da470b43f97fd0426b54bda739d7717fb28 for related change.

**Comment by jgreen210** [comment]
https://github.com/react-native-community/lottie-react-native/issues/453#issuecomment-474904070

**Comment by lorenc-tomasz** [comment]
Did you check if com.airbnb.android:lottie:2.8.0 works OK with animations? I have tested 2.6.0 and 2.6.1 and all my animations was broken or invisible :(

**Comment by abanik0831** [comment]
@kdawgwilk can you try upgrading to lottie android 3.0.0?

**Comment by greg7gkb** [comment]
We're waiting for this issue to be resolved as part of the upgrade path to RN 0.58 and subsequently to RN 0.59. It seems like there are a few interrelated issues so I just wanted to clarify that these are:
- Waiting for lottie-react-native to properly integrate with new androidx.* packages
- Upgrading underlying lottie libraries to version 3.x
- Support for android build tools 28.y

It seems that it all of these happen then the [Invalid Layer Save Flag Issue](https://github.com/react-native-community/lottie-react-native/issues/406) will also be resolved.

Please let me know if any of this is incorrect. Also, is there anything that I can do to help move things along?

**Comment by mvayngrib** [comment]
@kdawgwilk could you clarify why you closed? Are these changes no longer necessary?

**Comment by kdawgwilk** [comment]
This was only an issue when we were trying to use android X dependencies and so we reverted

**Comment by greg7gkb** [comment]
@kdawgwilk are you saying that lottie-react-native is now compatible with RN 0.59?

**Comment by greg7gkb** [comment]
Strictly speaking, lrn now supports RN 0.59, however the [Layer Save Flag](https://github.com/react-native-community/lottie-react-native/issues/406) issue remains and causes the following combination of build tools to crash our app at runtime:
* RN 0.59.8
* Android SDK/tools 28
* lottie-react-native 2.6.1

I'll follow up in that issue instead. Thanks.

**Pull request closed without merging**